pkg/alertmanager/statefulset.go

```diff
@@ -234,9 +234,17 @@ func makeStatefulSetSpec(a *monitoringv1.Alertmanager, config Config) (*appsv1.S
 	}
 
 	if *a.Spec.Replicas == 1 && !a.Spec.ForceEnableClusterMode {
-		amArgs = append(amArgs, "--cluster.listen-address=")
+		if a.Spec.ClusterListenLocal {
+			amArgs = append(amArgs, "--cluster.listen-address=127.0.0.1:9094")
+		} else {
+			amArgs = append(amArgs, "--cluster.listen-address=")
+		}
```

Comment:
In this case, we don't want to listen on gossip traffic at all.

```suggestion
		amArgs = append(amArgs, "--cluster.listen-address=")
```